### ISSUE TITLE
Question about pre-trained weights usability and performance on Hugging Face models

### ISSUE THREAD

**Issue by MichaelWu99-lab** [huggingface/lerobot#2221]
Hello,

I would like to ask whether the weights provided on Hugging Face (for example, under the lerobot author page) can be directly downloaded and used for inference, or if they must be fine-tuned before achieving reasonable performance.

When I directly load and evaluate the models (e.g., lerobot/smolvla_base or lerobot/pi05_libero_base), the performance appears extremely poor, almost random. I’m wondering if this is expected behavior or if I might have made a mistake in my setup.

Here’s the list of models I found on Hugging Face:

lerobot/smolvla_base
lerobot/pi05_base
lerobot/diffusion_pusht
lerobot/pi0_base
lerobot/pi05_libero_base
lerobot/act_aloha_sim_transfer_cube_human
lerobot/vqbet_pusht
lerobot/diffusion_pusht_keypoints
lerobot/act_aloha_sim_insertion_human
lerobot/pi0_libero_base
lerobot/pi05_libero_finetuned
lerobot/pi05_libero_finetuned_quantiles
lerobot/pi0_libero_finetuned

Are the *_base models supposed to be general pre-trained checkpoints that require downstream fine-tuning (e.g., on LIBERO), while the *_finetuned ones are ready for evaluation?

Thank you in advance for your clarification!

**Comment by fracapuano** [comment]
Ciao @MichaelWu99-lab 👋 Thank you for the question---I think it can be helpful for the community to discuss this.

**Short answer: no models on the Hub (aside from pi05) are intended to be used for zero-shot, language-conditioned inference.** That is, as of today no base model can effectively be used to perform a task right away, and you should always consider fine-tuning them on a given amount of demonstrations (unfortunately, no silver bullet there either---we have had good result with 25-50 demonstrations, but of course this depends on the task, robot, empirical conditions, etc...). Even pi05 might not perform well on your task if it is too dissimilar from its (large) amounts of training tasks, so you might need finetuning ([check out the docs](https://huggingface.co/docs/lerobot/en/pi05#training)).

Long answer: SmolVLA and PI0/05 are VLAs, intended to be large models that can perform multiple tasks generalizing across some distribution of tasks seen during training, conditionally only on language. The other models you have linked---ACT, Diffusion, and VQBeT---are all *single-task* models, which are developed to work out how to perform only _one_ task, and typically do not exploit language information as, in practice, they are used to learn one task, and one task only.
Caveat: think of the concept of _task_ as the composition of the actual task you want to perform (picking an object, slotting in a battery, etc.) and the _experimental conditions_ in which you performed that task while collecting the data the single task policy is trained on. You should not expect high generalization in out-of-distribution settings for large variations in any (or, even worse, both) dimensions! 

You can instead expect some sort of robustness to changes in the experimental conditions when using foundational models like SmolVLA because of the sheer difference of the dataset size used to train these models. With that being said, generalization is a tricky thing and is never easy to guarantee! Fine-tuning helps making your data more in-distribution for your model (in a really, really hand-wavy way but hey!)

If you're interested in learning more about any of these things, you may want to check out http://arxiv.org/abs/2510.12403. closing this, but feel free to re-open if it is still relevant :)

**Comment by dekelcohen** [comment]
Thanks. I have a followup question, please:
I have tested lerobot/pi05_libero_base and, as you mentioned, has 0 success rate on libero.

1) pi05, in particular, is reported to accomplish long horizon tasks at real homes. as Franka Panda is part of its training mixture, wouldn't one expect the pi05_base (or libero_base) to zeroshot success (no finetuning) on simple libero-spatial pick and place tabletop tasks ? Is there some mismatch with lerobot conventions (stats normalization, resolution ...) that prevents it from be used zeroshot ?

2) If finetune is required anyway - so pi05_libero_finetuned - works with lerobot - shouldn't it generalize to new Panda envs (not libero), with normal table top objects it must have seen in pretraining ? Otherwise, how did PhyIT achieved the reported results ?

**Comment by fracapuano** [comment]
I think your questions

> wouldn't one expect the pi05_base (or libero_base) to zeroshot success

and 

> shouldn't it generalize to new Panda envs (not libero), with normal table top objects

are both extremely valid _research_ questions that, fwiw, I strongly encourage you to pursue! Something I've learned (you might say in the hard way 😂) is that it's in general fairly complicated to have expectations regarding zero-shot performance and generalization. In plain twitter terms, I am trying really hard to be not too LLM-pilled bc the truth is that before ~GPT3 & the alike we used to use different models for summarization, translation, etc---i.e., we _didn't_ have zero shot generalization. In robot learning I am afraid we're not there (yet... 😉)

**Comment by LongJing0926** [comment]
I'd like to confirm that the pi05_libero_base huggingface model cannot be loaded using the from_pretrained method, right?
btw, I used the openpi source code to test the pi0.5_libero model on the libero dataset and it had a high success rate. Maybe you can follow this https://github.com/Physical-Intelligence/openpi/tree/main/examples/libero.

**Comment by fracapuano** [comment]
> I'd like to confirm that the pi05_libero_base huggingface model cannot be loaded using the from_pretrained method, right?

hey @LongJing0926 👋 Not sure I understand why you say that. Can you perhaps tell us more? Maybe link here some resources/error traces?

**Comment by dekelcohen** [comment]
> I'd like to confirm that the pi05_libero_base huggingface model cannot be loaded using the from_pretrained method, right? 

I successfully used it with the lerobot eval script, so it can be loaded. it may be correct that it doesn't simply call HF .from_pretrained,
